code/datums/gamemode/role/ninja.dm

```diff
@@ -18,10 +18,39 @@
 		name_ninja(antag.current)
 
 /datum/role/ninja/ForgeObjectives()
-	AppendObjective(/datum/objective/target/steal)
-	AppendObjective(/datum/objective/target/delayed/assassinate)
-	AppendObjective(/datum/objective/target/skulls)
-	AppendObjective(/datum/objective/escape)
+
+	if(cyborg_list.len)
+		AppendObjective(/datum/objective/target/killsilicons)
+	else
+		if(prob(70))
+			AppendObjective(/datum/objective/target/delayed/assassinate)
+		else
+			AppendObjective(/datum/objective/target/skulls)
+
+	if(ai_list.len)
+		AppendObjective(/datum/objective/killorstealAI)
+	else
+		AppendObjective(/datum/objective/target/steal)
+
+	if(player_list.len<=15 && prob(10))
```

Comment:
In retrospect this includes observers, dead, etc. so probably a better way is to pull the living playercount from the gamemode

Reply:
iterate & increment on some conditional?